googlecloud-plugins/src/main/java/co/cask/hydrator/plugin/source/BigQuerySource.java

```diff
@@ -0,0 +1,219 @@
+    ClassNotFoundException, InterruptedException {
+    Job job = JobUtils.createInstance();
+    Configuration conf = job.getConfiguration();
+    conf.set(MRBQ_JSON_KEY, sourceConfig.jsonFilePath);
+    conf.set(FSGS_JSON_KEY, sourceConfig.jsonFilePath);
+    conf.set(BigQueryConfiguration.PROJECT_ID_KEY, sourceConfig.projectId);
+    if (sourceConfig.importQuery != null) {
+      conf.set(BigQueryConfiguration.INPUT_QUERY_KEY, sourceConfig.importQuery);
+    }
+      conf.set(BigQueryConfiguration.TEMP_GCS_PATH_KEY, sourceConfig.tmpBucketPath);
+    conf.set("fs.gs.project.id", sourceConfig.projectId);
+    BigQueryConfiguration.configureBigQueryInput(conf, sourceConfig.fullyQualifiedInputTableId);
+    job.setOutputKeyClass(LongWritable.class);
+    job.setOutputValueClass(Text.class);
+    context.setInput(Input.of(sourceConfig.referenceName,
+                              new SourceInputFormatProvider(JsonTextBigQueryInputFormat.class, conf)));
+  }
+
+  @Override
+  public void transform(KeyValue<LongWritable, Text> input, Emitter<StructuredRecord> emitter) throws IOException {
+    getOutputSchema();
+    StructuredRecord record = jsonTransform(input.getValue());
+    emitter.emit(record);
+  }
+
+  private StructuredRecord jsonTransform(Text jsonText) {
+    Map<String, String> map;
+    map = GSON.fromJson(jsonText.toString(), MAP_STRING_STRING_TYPE);
+    StructuredRecord.Builder builder = StructuredRecord.builder(outputSchema);
+    for (Schema.Field field : outputSchema.getFields()) {
+      String fieldName = field.getName();
+      Schema schema = field.getSchema();
+      if (schema.getType() == Schema.Type.INT) {
+        builder.set(fieldName, Integer.parseInt(map.get(fieldName)));
+      } else {
+        builder.set(fieldName, map.get(fieldName));
+      }
+    }
+    return builder.build();
+  }
+
+  /**
+   * {@link PluginConfig} class for {@link BigQuerySource}
+   */
+  public static class BQSourceConfig extends PluginConfig {
+    private static final String IMPORT_QUERY = "importQuery";
+    private static final String PROJECT_ID = "projectId";
+    private static final String INPUT_TABLE_ID = "InputTableId";
+    private static final String JSON_FILE_PATH = "jsonFilePath";
+    private static final String TEMP_BUCKET = "tempBuketPath";
+    private static final String OUTPUT_SCHEMA = "outputSchema";
+    private static final String PROJECTID_DESC = "The ID of the project in Google Cloud";
+    private static final String TEMP_BUCKET_DESC = "the tempory google cloud storage directory to store the " +
+                                                   "intermediate result. Example: gs://bucketname/directoryname, " +
+                                                   "the directory should not be existed. User should delete this " +
+                                                   "directory afterward manually to avoid extra google storage charge.";
```

Comment:
```
private static final String TEMP_BUCKET_DESC = "The temporary Google Cloud Storage directory to be used for storing " +
                                               "the intermediate results. Example: 'gs://bucketname/directoryname'. " +
                                               "The directory should not already exist. Users should manually delete " +
                                               "this directory afterwards to avoid any extra Google Storage charges."